src/PDO/Builder.php

```diff
@@ -12,13 +12,17 @@ class Builder implements BuilderInterface
     protected const PROP_DATA_SOURCE_NAME = 'data_source_name';
     protected const PROP_USER_NAME        = 'user_name';
     protected const PROP_PASSWORD         = 'password';
+    protected const PROP_PORT             = 'port';
     protected const PROP_OPTIONS          = 'options';
     protected $_pdo;
 
     public function getPdo(): \PDO
     {
         if ($this->_pdo === null) {
             $dsn = $this->_getDataSourceName();
+            if ($this->_hasPort() && $this->_getPort() !== 0){
+                $dsn = $dsn . sprintf(';port=%d', $this->_getPort());
```

Comment:
non-blocker:
```suggestion
                $dsn = sprintf('%s;port=%d', $this->_getDataSourceName(), $this->_getPort());
```
and I would remove the above 
```php
$dsn = $this->_getDataSourceName();
```
and place it in an `else` instead of changing the value of `$dsn`